### PR TITLE
Added a debug option to dump serial traffic

### PR DESCRIPTION
**Pull request by baffo32**
```
Sending soft reset
Waiting for '$$$'
Debug bytes:
>>>  76                                                |v|
Debug bytes:
0<<  4f 70 65 6e 42 43                                 |OpenBC|
Debug bytes:
0<<  49 20 56 33 20 53 69 6d 75  6c 61 74 6f 72 20 4f  |I V3 Simulator O|
0<<  6e 20 42 6f 61 72 64 20 41  44 53 31 32 39 39 20  |n Board ADS1299 |
0<<  44 65 76 69 63 65 20 49 44  3a 20 30 78 31 32 33  |Device ID: 0x123|
0<<  34 35 20 20 4c 49 53 33 44  48 20 44 65 76 69 63  |45  LIS3DH Devic|
0<<  65 20 49 44 3a 20 30 78 33  38 34 32 32 20 24 24  |e ID: 0x38422 $$|
0<<  24                                                |$|
Debug bytes:
>>>  62                                                |b|
Debug bytes:
3<<  a0 00 00 01 b2 ff ff d9 00  00 18 00 00 59 00 04  |.............Y..|
3<<  62 00 03 e0 00 01 00 00 01  2a 00 00 00 00 00 00  |b........*......|
3<<  c0 a0 01 00 04 90 00 02 4d  ff fb 31 00 00 e3 ff  |........M..1....|
3<<  fc c4 00 02 33 ff fe d5 00  07 6c 00 00 00 00 00  |....3.....l.....|
3<<  00 c0 a0 02 ff ff 59 00 03  79 00 01 7c ff f9 9e  |......Y..y..|...|
3<<  ff f9 86 ff fb 3c ff fd 06  00 02 ca 00 00 00 00  |.....<..........|
3<<  00 00 c0 a0 03 00 03 6b ff  fe 2e ff fe cb 00 00  |.......k........|
3<<  13 ff fd 22 ff fe 6d ff f9  68 00 00 d3 00 00 00  |..."..m..h......|
3<<  00 00 00 c0                                       |....|
stop streaming
Debug bytes:
>>>  73                                                |s|
```

### CODE REVIEW

**Comment by baffo32** [comment]
whoops forgot coverage again

File: openBCIBoard.js [hunk 1]
```diff
@@ -110,6 +111,8 @@ function OpenBCIFactory () {
   *
   *     - `verbose` {Boolean} - Print out useful debugging events
   *
+  *     - `debug` {Boolean} - Print out a raw dump of bytes sent and received
```
Comment (aj-ptw) [inline openBCIBoard.js]: Please add `(Default `false`)` additionally, I now see that `verbose` is missing the `(Default `false`)`, please add there too.

File: test/openBCIBoard-test.js [hunk 2]
```diff
@@ -417,6 +417,40 @@ describe('openbci-sdk', function () {
         }).catch(err => done(err));
     });
   });
+  describe('#debug', function () {
+    before(function (done) {
+      ourBoard = new openBCIBoard.OpenBCIBoard({
+        simulate: true,
+        fragmentation: k.OBCISimulatorFragmentationNone,
```
Comment (aj-ptw) [inline test/openBCIBoard-test.js]: extraneous line

**Comment by aj-ptw** [comment]
This may be my favorite new feature haha

**Comment by baffo32** [comment]
fixed

**Comment by baffo32** [comment]
could travis be having network issues? there's also a failing network-related task on 1.4.0 branch

**Comment by aj-ptw** [comment]
@baffo32 please fix merge conflict and this is g2g

**Comment by baffo32** [comment]
merges clean now

**Comment by aj-ptw** [comment]
Great work @baffo32!! This feature is epic!

**Pull request merged**